iceoryx_utils/test/moduletests/test_cxx_string.cpp

```diff
@@ -905,3 +1117,972 @@ TYPED_TEST(stringTyped_test, StringWithContentIsNotEmtpy)
+    string<100> testString2(TruncateToCapacity, testStdString.c_str(), 7);
+    res = testString2.find("e\0ry", 0);
+    EXPECT_THAT(res.has_value(), Eq(true));
+    EXPECT_THAT(res.value(), Eq(2));
+}
+
+TEST(String100, FindNotIncludedStringLiteralFails)
+{
+    string<100> testString("Kernfusionsbaby");
+    auto res = testString.find("abc");
+    EXPECT_THAT(res.has_value(), Eq(false));
+
+    res = testString.find("abc", 50);
+    EXPECT_THAT(res.has_value(), Eq(false));
+}
+
+TEST(String100, FindSTDStringInNotEmptyStringWorks)
+{
+    string<100> testString("R2-D2");
+    std::string testStdString = "2";
+    auto res = testString.find(testStdString);
+    EXPECT_THAT(res.has_value(), Eq(true));
+    EXPECT_THAT(res.value(), Eq(1));
+
+    res = testString.find(testStdString, 1);
+    EXPECT_THAT(res.has_value(), Eq(true));
+    EXPECT_THAT(res.value(), Eq(1));
+
+    res = testString.find(testStdString, 2);
+    EXPECT_THAT(res.has_value(), Eq(true));
+    EXPECT_THAT(res.value(), Eq(4));
+}
+
+TEST(String100, FindNotIncludedSTDStringFails)
+{
+    string<100> testString("Kernfusionsbaby");
+    std::string testStdString = "abc";
+    auto res = testString.find(testStdString);
+    EXPECT_THAT(res.has_value(), Eq(false));
+
+    res = testString.find(testStdString, 0);
+    EXPECT_THAT(res.has_value(), Eq(false));
+
+    res = testString.find(testStdString, 50);
+    EXPECT_THAT(res.has_value(), Eq(false));
+}
+
+/// @note template <typename T>
+/// iox::cxx::optional<uint64_t> find_first_of(const T& t, uint64_t pos = 0) const noexcept
+TYPED_TEST(stringTyped_test, FindFirstOfFailsForEmptyStringInEmptyString)
+{
+    using myString = typename TestFixture::stringType;
+    constexpr auto STRINGCAP = myString().capacity();
+    string<STRINGCAP + 5> testString;
+    auto res = this->testSubject.find_first_of(testString);
+    EXPECT_THAT(res.has_value(), Eq(false));
+
+    res = this->testSubject.find_first_of("");
+    EXPECT_THAT(res.has_value(), Eq(false));
+
+    std::string testStdString;
+    res = this->testSubject.find_first_of(testStdString);
+    EXPECT_THAT(res.has_value(), Eq(false));
+}
+
+TYPED_TEST(stringTyped_test, FindFirstOfForStringInEmptyStringFails)
+{
+    using myString = typename TestFixture::stringType;
+    constexpr auto STRINGCAP = myString().capacity();
+    string<STRINGCAP + 5> testString("a");
+    auto res = this->testSubject.find_first_of(testString);
+    EXPECT_THAT(res.has_value(), Eq(false));
+
+    res = this->testSubject.find_first_of("a");
+    EXPECT_THAT(res.has_value(), Eq(false));
+
+    std::string testStdString = "a";
+    res = this->testSubject.find_first_of(testStdString);
+    EXPECT_THAT(res.has_value(), Eq(false));
+}
+
+TEST(String100, FindFirstOfForStringInNotEmptyStringWorks)
+{
+    string<10> testString("R2-D2");
+    string<100> substring1("2");
+    auto res = testString.find_first_of(substring1);
+    EXPECT_THAT(res.has_value(), Eq(true));
```

Comment:
ASSERT_THAT, I will stop marking this issue but further tests are affected as well. Always ASSERT_THAT when we try to access the value afterwards.

Reply:
done